rr_arduino/imu.ino

```diff
@@ -42,29 +42,32 @@ void imu_readings()
   //Get a new sensor event
   sensors_event_t event; 
   bno.getEvent(&event);
+  imu::Vector<3> acc = bno.getVector(Adafruit_BNO055::VECTOR_LINEARACCEL);
+  imu::Vector<3> gyro= bno.getVector(Adafruit_BNO055::VECTOR_GYROSCOPE);
+  imu::Vector<3> mag= bno.getVector(Adafruit_BNO055::VECTOR_MAGNETOMETER);
 
   //Get current time
   ros::Time current_time=ros::Time::now();
   //assigning values for imu message
-  imu_msg.orientation.x=event.orientation.x;
-  imu_msg.orientation.y=event.orientation.y;
-  imu_msg.orientation.z=event.orientation.z;
+  imu_msg.orientation.x=event.orientation.x();
```

Comment:
this is not quite correct the x,y and z fields here refer to quaternions and not euler angle representations

Reply:
this is only for orientation

Reply:
okay.

Reply:
This website has a sections where this is done. https://www.allaboutcircuits.com/projects/bosch-absolute-orientation-sensor-bno055/